### PR TITLE
[Windows] use dynamic temp path

### PR DESCRIPTION
**Pull request by carlogrisetti**
Use dynamic temp path (this fixes trying to create a lockfile under Windows)

### CODE REVIEW

**Comment by carlogrisetti** [comment]
@iycheng would be nice to include prior to 1.7.0 release (just tagging you since I see you're active rn)

**Comment by carlogrisetti** [comment]
@iycheng would be nice to include prior to 1.7.0 release (tagging you since I see you active rn)

File: python/ray/autoscaler/_private/local/config.py [hunk 1]
```diff
@@ -69,12 +71,19 @@ def prepare_manual(config: Dict[str, Any]) -> Dict[str, Any]:
     return config
 
 
+def get_temp_path() -> str:
+    ray_temp = os.path.join(tempfile.gettempdir(), "ray")
+    if not os.path.exists(ray_temp):
+        os.makedirs(ray_temp)
+    return ray_temp
```
Comment (ijrsvt) [inline python/ray/autoscaler/_private/local/config.py:78]: Hey @carlogrisetti This looks like a good thing to change! I think the functionality of `get_temp_path` is already present in https://github.com/ray-project/ray/blob/1efcf5c3d5e304db625a085cd40ff003ab99dc63/python/ray/_private/utils.py#L48

Could you refactor to directly use that function?

**Comment by ijrsvt** [comment]
cc @DmitriGekhtman

**Comment by carlogrisetti** [comment]
@DmitriGekhtman oh yes I know Windows isn't officially supported, but I think we could be really near it having to work well. I had good results with 1.4.x in the past, and I now am getting back to using Ray.
I prefer windows as it's my default developing environment, and I thus don't need any separate docker environment or something else to work with, since I also have access to numerous windows machines that are mostly idle all the time.
Having a cluster set up on windows machines could mean I can leverage those resources.

I used the get_ray_temp_dir() function, to keep all ray-related things in one place (that was the same reasoning behind my PR, and I ended up using exactly the same path). I suppose it's what you intended when saying to use get_user_temp_dir(), but if there is a preference in using the "raw" user temp dir, I'll change this again.

**Comment by ijrsvt** [comment]
@DmitriGekhtman Are you good with moving local mode into `/<temp_dir>/ray/` as opposed to `/<temp_dir>/`.

**Comment by DmitriGekhtman** [comment]
> @DmitriGekhtman Are you good with moving local mode into `/<temp_dir>/ray/` as opposed to `/<temp_dir>/`.

That's a good catch. Probably it's fine.

**Review by DmitriGekhtman — APPROVED** [review]
Thanks, looks good. Will take a closer look at CI before merging.

**Comment by pcmoritz** [comment]
This looks good to me now, just triggered the windows build too.

@DmitriGekhtman Any last thoughts on this before we merge it?

**Pull request merged**